x/lend/keeper/keeper.go

```diff
@@ -159,7 +175,20 @@ func (k Keeper) LendAsset(ctx sdk.Context, lenderAddr string, AssetID uint64, Am
 	}
 
 	if k.HasLendForAddressByAsset(ctx, lenderAddr, AssetID, PoolID) {
-		return types.ErrorDuplicateLend
+		// if a lend position is opened by the user for same asset in a pool, and when the user tries to lend again in that case
+		// we will deposit that asset and increase user's previous lend position.
+		// Steps:
+		// Get the lend ID of previous lend position
+		// Call Deposit function
+		lendID := k.ReturnLendID(ctx, lenderAddr, AssetID, PoolID)
+		_, found := k.GetLend(ctx, lendID)
+		if !found {
+			return types.ErrLendNotFound
+		}
```

Comment:
No need of validating lendID again, if some bool like `found` can be added in the method `ReturnLendID` itself where they are already being accessed.

Reply:
Resolved